### ISSUE TITLE
Multi-threaded clustering

### ISSUE THREAD

**Issue by jcoupey** [VROOM-Project/vroom#91]
The current CVRP heuristic implementation is detailed in #74 and implemented in #75. Bottleneck for overall computing time is that several clustering parameters are used sequentially and we then pick up the one that looks the more promising. Clustering steps are independent so running them in parallel should be straightforward for a nice gain.

The same applies to the TSP solving processes (one per cluster). As they are already quite fast, the expected gain is much smaller. It might still make a noticeable difference when clusters start to hold a large number of jobs.

**Comment by jcoupey** [comment]
Comparing running the clustering part with `-t 1` and in parallel with `-t 4`: overall computing time for the current heuristic on classes A, B, E, F, M, P, X from [CVRPLIB](http://vrp.atd-lab.inf.puc-rio.br/index.php/en/) goes down from ~220 seconds to ~100 seconds.

**Comment by jcoupey** [comment]
After the clustering selection, we solve one TSP per vehicle. Doing that in parallel won't make a difference in most cases. For example on the above mentioned instances, the average number of job per vehicle tops at 112 and is below 14 on average. As a result, the TSP solving phase ranges from 0 to 11 ms only (!!) on my laptop.

However, I implemented multi-threaded TSP solving anyway because it will still be useful in some edge cases. For example, I tested on an instance from TSPLIB with 5934 jobs that I adjusted to use 2 vehicles. Then the time spent solving the 2 TSP drops from ~30 seconds with `-t 1` to ~20 seconds with `-t 4` (on 4de846b).
